arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorHolder.java

```diff
@@ -142,6 +152,12 @@ public ConstantVectorHolder(int numRows, T constantValue) {
       this.constantValue = constantValue;
     }
 
+    public ConstantVectorHolder(Types.NestedField icebergField, int numRows, T constantValue) {
+      super(icebergField);
```

Comment:
Each `VectorHolder` actually has `icebergField` but we were not setting it for constants.